### ISSUE TITLE
operator should ensure all serving components are remain installed as long as CR is present

### ISSUE THREAD

**Issue by yu2003w** [knative/serving-operator#81]
/kind feature

**Branch:** 
/master
/HEAD

**Current Behavior:**
Currently after installing Knative-serving with serving-operator, all components such as controller, activator, autoscaler and so on are deployed but not become 'available' due to some other reasons.
For example, image pull secret is missing.
Then delete the deployments in namespace knative-serving, operator will not reinstall the deployments again.

**Expected Behavior:**
operator could reinstall deployments as long as CR still present.

 **Detailed Procedure**
1, remove image pull secret from namespace knative-serving
2, update serving_v1alpha1_knativeserving_cr.yaml to apply customized images
```
spec:
  registry:
    override:
      activator: icpdev.icp:8500/imgrepo/serving-activator-amd64:v0.6.1
      autoscaler: icpdev.icp:8500/imgrepo/serving-autoscaler-amd64:v0.6.1
      controller: icpdev.icp:8500/imgrepo/serving-controller-amd64:v0.6.1
      networking-certmanager: icpdev.icp:8500/imgrepo/serving-certmanager-amd64:v0.6.1
      networking-istio: icpdev.icp:8500/imgrepo/serving-istio-amd64:v0.6.1
      webhook: icpdev.icp:8500/imgrepo/serving-webhook-amd64:v0.6.1
```
2, launch serving-operator to install knative-serving
3, delete deployments in namespace knative-serving.
deleted deployments are not created again.

**Comment by savitaashture** [comment]
@yu2003w can you give me little more detail like where you run operator and which version.
Because i verified in openshift its working and operator able to recreate deployment if its deleted.

**Comment by yu2003w** [comment]
@savitaashture I deployed serving-operator with master branch and deploy on original k8s cluster. I don't have a working openshift cluster to try now.

**Comment by savitaashture** [comment]
@yu2003w to make sure i just did following steps
1. Installed minikube
2. Installed istio following https://github.com/knative/serving-operator#prerequisites
3. created
    ```
    kubectl apply -f config/crds/serving_v1alpha1_knativeserving_crd.yaml
    ko apply -f config/
    kubectl create -f config/crds/serving_v1alpha1_knativeserving_cr.yaml
    ```
4. All deployments running in `knative-serving` namespace
```
kubectl get pods -n knative-serving
NAME                                     READY   STATUS    RESTARTS   AGE
activator-5ccf59c56-6hrgc                1/1     Running   0          79s
autoscaler-7f55589d4b-h4r57              1/1     Running   0          79s
controller-77cd5ffdc4-bbcfj              1/1     Running   0          74s
networking-certmanager-554f949d5-r54wh   1/1     Running   0          73s
networking-istio-5476685c6f-4cnvf        1/1     Running   0          73s
webhook-5f668cf9f8-m8mbw                 1/1     Running   0          72s
```
5. Deleted webhook deployment then serving-operator recreated webhook
```
kubectl get deployment -n knative-serving
NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
activator                1/1     1            1           2m37s
autoscaler               1/1     1            1           2m37s
controller               1/1     1            1           2m32s
networking-certmanager   1/1     1            1           2m31s
networking-istio         1/1     1            1           2m31s
webhook                  1/1     1            1           12s
```

Did you follow this way or any different steps because i can see its working

**Comment by yu2003w** [comment]
Sorry for the misleading description in the original issue. I updated the description.
I tried again and reproduced the issue.

**Comment by yu2003w** [comment]
Fixed by PR#83
